### PR TITLE
docs: add error-overlay for sandbox

### LINKED ISSUE #3922: docs: trpc.io homepage demo is not working
### Area of Improvement

The demo on the tPRC homepage is not working on FF 109.0. I'm not getting any syntax highlighting, and the preview shows an error message: 

> Unable to run Embedded Project
> 
> Looks like this project is being embedded without proper isolation headers.
> 
> [Read more about browser support for embedded projects](https://developer.stackblitz.com/platform/webcontainers/browser-support#embedding)

Works fine in edge.

### Link to related docs

https://trpc.io/

### Additional information

_No response_

### 👨‍👧‍👦 Contributing

- [ ] 🙋‍♂️ Yes, I'd be down to file a PR implementing the suggested changes!

### PR DESCRIPTION
**Pull request by juliusmarminge**
Closes #3922

## 🎯 Changes

What changes are made in this PR? Is it a feature or a bug fix?

I made it dismissable so that they can still see the code if they want.

Test it by opening this link in a Chromium vs Safari browser: https://www-git-sandbox-overlay-trpc.vercel.app/#try-it-out

## ✅ Checklist

- [ ] I have followed the steps listed in the [Contributing guide](https://github.com/trpc/trpc/blob/main/CONTRIBUTING.md).
- [ ] If necessary, I have added documentation related to the changes made.
- [ ] I have added or updated the tests related to the changes made.

### CODE REVIEW

**Comment by juliusmarminge** [comment]
cc @c-ehrlich thoughts?

File: www/src/pages/index.tsx [hunk 1]
```diff
@@ -61,6 +61,41 @@ const useQueryParam = (key: string) => {
...
+    <div className="absolute inset-0 z-10 flex flex-col items-center justify-center gap-2 overflow-hidden bg-zinc-900">
+      <div className="flex flex-col items-center justify-center gap-4">
+        <div className="text-5xl font-bold text-zinc-900 dark:text-zinc-100">
+          ⚠️
+        </div>
+        <div className="text-xl font-bold text-zinc-900 dark:text-zinc-100">
+          Heads up!
+        </div>
+        <div className="w-full max-w-xl text-lg font-medium text-zinc-700 dark:text-zinc-200">
+          WebContainers require modern Web APIs that likely isn&apos;t supported
```
Comment (c-ehrlich) [inline www/src/pages/index.tsx:75]:
```suggestion
          WebContainers require modern Web APIs that likely aren&apos;t supported
```

File: www/src/pages/index.tsx [hunk 2]
```diff
@@ -61,6 +61,41 @@ const useQueryParam = (key: string) => {
...
+          by your browser. Either open this page in a different browser, or
```
Comment (c-ehrlich) [inline www/src/pages/index.tsx:76]: Maybe just say Chrome / Chromium? Or is there any other browser/engine with good enough support?

**Review by KATT — COMMENTED** [review]
Ummmm.... can't we post an issue in StackBlitz instead? They should be able to show an error message and keep that maintained to the browsers they support

File: www/src/pages/index.tsx [hunk 3]
```diff
@@ -70,6 +105,17 @@ function TryItOut() {
     sandboxes.find((it) => it.id === param) ?? sandboxes[0];
   const selectedId = selected.id;
 
+  const [deviceSupported, setIsSupported] = React.useState(() => {
+    if (typeof window === 'undefined') return false;
+
+    const ua = window.navigator.userAgent;
+    console.log({ ua });
```
Comment (KATT) [inline www/src/pages/index.tsx:112]:
```suggestion
```

**Pull request merged**